src/main/java/org/cactoos/io/DigestOf.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.cactoos.Bytes;
+import org.cactoos.Input;
+
+/**
+ * Checksum calculation of {@link Input}.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.28
+ */
+public final class DigestOf implements Bytes {
+
+    /**
+     * The input.
+     */
+    private final Input source;
+
+    /**
+     * The buffer size.
+     */
+    private final int size;
+
+    /**
+     * The algorithm.
+     */
+    private final String algorithm;
+
+    /**
+     * Ctor.
+     * @param input The input
+     */
+    public DigestOf(final Input input) {
+        this(input, "SHA-256");
```

Comment:
using plain strings to represent/select settings was always a major pain for me (was ist "sha-256" or "SHA-256" or was ist "sha256"?. I understand why it can be beneficial to allow plain strings, but could we also provide a Ctor which takes an enum? This makes it easier to select a algorithm. By adding a Ctor which takes an enum we do not prevent ppl to use plain strings if they want.

Provide a few ENUM representing commonly used and available algorithms like MD5/SHA256

Reply:
@Timmeey Yes, there is this problem, but if I'm not wrong, Yegor don't like `enums`. @yegor256 WDYT? Should I put enums in this code? (Maybe a post about why enums are a mistake would be nice.)

Reply:
Enums are just an easy way of showing the developer which values are available. In my opinion it is WAY worse to force the developer to look up possible values by going through the sourcecode.

If we don't want enums, we need classes like `SHA256DigestOf` `MD5DigestOf` and then one where you have to provie the algorithm name. THis could then be done by having an abstract DigestOf class with the logic inside, and concrete classes that only provide their appropriate Algorithm string to the abstract DigestOf class. It's either specific classes or enums.

Plain strings as the only way to select the algorithm are just a pain.

Reply:
@Timmeey I agree with this solution. I think it better than enums. I'll provide them. Thanks!